### PR TITLE
Make the upload look like the mockups

### LINKED ISSUE #51: Change style of file upload widget on work form
See [mock-up](https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420954309). What we have currently works but it's vanilla Dropzone. We need to get it closer to the style in the H2 design.

- [ ] #119
- [ ] #120

### LINKED ISSUE #119: Add file label to the form
When a file is uploaded, a new label field should be displayed:

Screen Shot 2020-10-09 at 8 22 36 AM

This will be stored in a new database field.

### LINKED ISSUE #120: Add hide file button to the form
I can choose not to publish any individual files uploaded, and confirm they are not displayed in PURL

You can choose this option with the checkbox on the right:

Screen Shot 2020-10-09 at 8 22 36 AM

This should be stored in a new database field.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #119 
Fixes #120 
Fixes #51

## How was this change tested?

Screen Shot 2020-10-27 at 12 35 16 PM

### CODE REVIEW

File: app/javascript/controllers/dropzone_controller.js [hunk 1]
```diff
@@ -72,12 +72,15 @@ export default class extends Controller {
...
-    return this.data.get("addRemoveLinks") || true;
+    return this.data.get("addRemoveLinks") || false;
   }
 }
 
 class DirectUploadController {
+  // source is the input element
+  // file is the File object from dropzone
   constructor(source, file) {
+    this.count = Math.floor(Math.random() * 1_000_000_000)
```
Comment (mjgiarlo) [inline app/javascript/controllers/dropzone_controller.js:83]: What is this count for?

File: app/models/work.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class Work < ApplicationRecord
   has_many :contributors, dependent: :destroy
   has_many :related_links, dependent: :destroy
   has_many :related_works, dependent: :destroy
-  has_many_attached :files
+  has_many :attached_files, dependent: :destroy
```
Comment (mjgiarlo) [inline app/models/work.rb:12]: Can you say more about this change?

Comment (mjgiarlo) [reply app/models/work.rb:12]: Nevermind, I now see it's so you can hang the label and hide attrs off of attached files.

File: spec/models/attached_file_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+RSpec.describe AttachedFile, type: :model do
+  subject(:attached_file) { build(:attached_file, :with_file) }
+
+  it 'has a label' do
+    expect(attached_file.label).to be_present
+  end
+
+  it 'has a hide bit' do
+    expect(attached_file).not_to be_hide
```
Comment (mjgiarlo) [inline spec/models/attached_file_spec.rb:14]: `expect(armor).not_to be_hide`

I wonder if `be_hidden` would automagically work here?

**Review by mjgiarlo — APPROVED** [review]
Some questions, but looking good.

File: app/javascript/controllers/dropzone_controller.js (hunk 1 above)
Comment (jcoyne) [reply app/javascript/controllers/dropzone_controller.js:83]: Need a unique identifier when you submit rails nested attributes.  It's the same sort of thing that's going on here (but timestamp doesn't have enough entropy in this case): https://github.com/sul-dlss/happy-heron/blob/main/app/javascript/controllers/nested_form_controller.js#L8

File: spec/models/attached_file_spec.rb (hunk 3 above)
Comment (jcoyne) [reply spec/models/attached_file_spec.rb:14]: Nope:
```
 Failure/Error: expect(attached_file).not_to be_hidden
       expected #<AttachedFile id: nil, label: "MyString", hide: false, work_id: nil, created_at: nil, updated_at: nil> to respond to `hidden?`
```

Comment (mjgiarlo) [reply spec/models/attached_file_spec.rb:14]: :rage:

**Pull request merged**